controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -199,6 +216,27 @@ func (r *StorageClassRequestReconciler) reconcilePhases() (reconcile.Result, err
 				break
 			}
 		}
+
+		// check if a cephblockpool resource exists for the desired storageconsumer and storageprofile.
+		if r.cephBlockPool.Name == "" {
+			cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
+			listOptions := &client.MatchingLabels{
+				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+				controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
+			}
+			if err := r.list(cephBlockPoolList, client.InNamespace(r.OperatorNamespace), listOptions); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// if we find only one CephBlockPool with our query, we're good
+			// if we find more than one, we can't determine which one to select, so error out
+			if len(cephBlockPoolList.Items) == 1 {
```

Comment:
Can we handle the case we didn't find here (len == 0) instead of relegating it to the end?
This way we have symmetry

```
if len == 0 {
} else if len == 1 {
} else { // len >1
}